types/protobuf-fieldmask/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// Type definitions for protobuf-fieldmask 1.2
+// Project: http://github.com/kibertoad/protobuf-fieldmask
+// Definitions by: Jorge Yero Salazar <https://github.com/jyeros>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export type WithFieldMask<T> = { [K in keyof T]?: WithFieldMask<T[K]> | undefined };
+/**
+ * Creates a new object that copies fields present in field mask from specified source object
+ * @param sourceObject - object to apply field mask to
+ * @param fieldMask
+ * @returns new object created by applying field mask on source object or original entity if source is not an object
+ */
+export function applyFieldMask<T>(sourceObject: T, fieldMask: string[]): WithFieldMask<T>;
```

Comment:
The `fieldMask` parameter here can be marked `readonly` since it's never modified:

```suggestion
export function applyFieldMask<T>(sourceObject: T, fieldMask: readonly string[]): WithFieldMask<T>;
```